### PR TITLE
Added functions `G` and `DG` and made the code more usuable

### PR DESCRIPTION
**Pull request by abhikandoi2000**

### CODE REVIEW

File: cocktail_party.m [hunk 1]
```diff
@@ -1,9 +1,9 @@
-[x1, Fs1] = wavread('~/Academics/Sem4/EE214/source2.wav');
-[x2, Fs2] = wavread('~/Academics/Sem4/EE214/source4.wav');
+[x1, Fs1] = wavread('mic1.wav');
+[x2, Fs2] = wavread('mic2.wav');
+
 m = size(x1,1);
 n = 2;
-A = randn(n, n);
```
Comment (kmspriyatham) [inline cocktail_party.m:5]: Don't remove this line and the next as it models a linear combination of the sources.

**Comment by kmspriyatham** [comment]
Are you going to use this code for something?

**Comment by abhikandoi2000** [comment]
I am trying to learn different techniques for source separation. Thought I should give it a try.
Why do you ask by the way?

**Comment by kmspriyatham** [comment]
To see of the code I wrote is useful to anyone at all. :D

**Comment by abhikandoi2000** [comment]
Yeah. Nice work there buddy. :+1:

**Pull request merged**